src/connector/src/base.rs

```diff
@@ -62,13 +62,19 @@ pub struct ConnectorState {
     pub end_offset: String,
 }
 
+#[derive(Debug, Clone)]
+pub enum ConnectorStateV2 {
```

Comment:
This is too dirty. We definitely need another implementation or another name later.